Rocketfuel quest refactor

HFPTetraUro opened the pull request:
Refactor rocket fuel quests

<img width="156" height="653" alt="image" src="https://github.com/user-attachments/assets/80c7cea4-4423-4f41-833c-14c53445c0fe" />

* Have a clear step for CBD in the quest line
* Add missing rewards and re-tier existing rewards
* Replace checkbox quests and optional retrieval with regular retrieval
* Added trigger quest skips to replace the checkbox quests, skipping all quests up that fuel tier after holding 32 cells, and mention the skips exist and how to perform them
* Update description to keep continuity, fix grammar, clarify, etc.
* All skips require checking the box on "A Better Rocket Fuel" quest, the last one also requires unlocking the ZPM tier
* Update quest prerequisites in some places

UltraProdigy reviewed (commented):
Is there a reason for the coin reward removal from a bunch of these quests? We love coins around here

Nikolay-Sitnikov commented on config/betterquesting/DefaultQuests/Quests/SpaceRace-AAAAAAAAAAAAAAAAAAAAIg==/RocketFuel-AAAAAAAAAAAAAAAAAAADOw==.json:
```diff
@@ -8,7 +8,7 @@
     "betterquesting:10": {
       "autoClaim:1": 0,
-      "desc:8": "For every flight the rocket needs some fuel. So refine some fuel for your rocket.\n\nFor your T1/2 rockets you have 2 options. You can either use Cetane-Boosted Diesel or make RP-1 rocket fuel. We will have a look at RP-1 rocket fuel in the next quest.\n\nYou\u0027ll need something with more oomph for the later rockets.\n\nFor the moon, a full rocket leaving from the OW should ha ...
+      "desc:8": "For every flight the rocket needs some fuel. So refine some fuel for your rocket.\nThis quest chain will guide you through the progression of rocket fuels.\n\nFor your T1/2 rockets you have 2 options. You can either use Cetane-Boosted Diesel or make RP-1 rocket fuel.\n\nFor the moon, a full rocket leaving from the Overworld should have enough fuel left to make it back. But you mi ...
```
```suggestion
      "desc:8": "For every flight the rocket needs some fuel. So refine some fuel for your rocket.\nThis quest chain will guide you through the progression of rocket fuels.\n\nFor your T1 & T2 rockets you have 2 options. You can either use Cetane-Boosted Diesel or make RP-1 rocket fuel.\n\nFor the moon, a full rocket leaving from the Overworld should have enough fuel left to make it back, but you might want to take extra just in case.\n\n[note]Rocket Fuel usage is not based on the destination nor the type of fuel used. Rather, it is based on the tier of the rocket, with higher tiers both storing more fuel, and consuming less. Also, the gravity of the planet affects how much fuel is used, with lower gravity locations using less fuel to launch.[/note]",
```

Nikolay-Sitnikov commented on config/betterquesting/DefaultQuests/Quests/SpaceRace-AAAAAAAAAAAAAAAAAAAAIg==/Naphthalene-AAAAAAAAAAAAAAAAAAAKhg==.json:
```diff
@@ -8,7 +8,7 @@
     "betterquesting:10": {
       "autoClaim:1": 0,
-      "desc:8": "To make Naphthalene, combine your Coal Oil Tar with Sulfuric Acid to get Sulfuric Coal Tar Oil.\n\nThen distill it to get Naphthalene. Seem easy? It gets harder.",
+      "desc:8": "To make Naphthalene, combine your Coal Tar Oil with Sulfuric Acid to get Sulfuric Coal Tar Oil.\n\nThen distill it to get Naphthalene. Seems easy? It gets harder.",
```
```suggestion
      "desc:8": "To make Naphthalene, combine your Coal Tar Oil with Sulfuric Acid to get Sulfuric Coal Tar Oil, then distill it to get Naphthalene. Seems easy? It gets harder.",
```

Why the line break in the middle of a sentence?

HFPTetraUro commented:
> I think this PR needs to be revised again. It's okay to change the text or the loot. However, the coins and loot bags should remain as they are and must not be removed. If you redesign the quests, the symmetry must also be correct. Otherwise, the whole thing looks rather unattractive (see image).

see my other reply for the loot changes. If by symmetry you mean the layout of the quests, I didn't touch it except the 3 starting quests which I made sure to align. I don't mind moving around the later quests to make it symmetrical, though.

HFPTetraUro commented on config/betterquesting/DefaultQuests/Quests/SpaceRace-AAAAAAAAAAAAAAAAAAAAIg==/ABetterRocketFue-qz7VHLFvQx6gDw3KZDIj0g==.json:
```diff
@@ -1,18 +1,18 @@
     "betterquesting:10": {
       "autoClaim:1": 0,
-      "desc:8": "If you want to go further than Mars you need a better rocket fuel. From here on out there is only one choice. The Chemical Plant is at the heart of the production. If you don\u0027t have one yet, you should have a look at it. It has 4 separate tier system and might be a bit over whelming at first, but there is a minecraft book that can tell you everything about it. Also check out ...
+      "desc:8": "If you want to go further than Mars you need a better rocket fuel. From here on out there is only one choice. The Chemical Plant is at the heart of the production. If you don\u0027t have one yet, you should have a look at it. It has 4 separate tiering systems and might be a bit overwhelming at first, but there is a Minecraft book that can tell you everything about it. \n\n[note]A ...
```
Not sure what you mean by 2 choices? Anyways probably should change the note to not start with "also" if we're omitting the book, since thay quest is gonna be your main source of info

HFPTetraUro commented:
> > > Is there a reason for the coin reward removal from a bunch of these quests? We love coins around here
> > 
> > 
> > It was from a lot of the later quests for the green rocket fuel where they gave you 50 coins and a choice between random loot and an HV lootbag, for a fuel supposedly tiered at ZPM, so the lootbag is useless and a bunch of quests before give more coins so it made no sense.
> > I added HV lootbags to the first quests in the chain up until the one that requires Palladium since you can technically complete them in HV and before actually unlocking the fuel.
> > But at that point it felt wrong to reward appropriate tiered lootbags for every step so I put the lootbags in the final quest for each chain and kept the loot in every step simple unless that step had a harder requirement (e.g. the Laurenium casing).
> > But I think it's fine to add coins to every step, not sure why I didn't tbh.
> 
> Just let the lootbags and coins in the quest and unchanged. i can later check the amount and correct it if they feel unbalanced. I will go over quests for sure before stable one more time. Even if you not change the Quest icons it would be nice bring them in a symetric order so they have to look more clean.

Why not fix it now? Why have quests that you do in ZPM that give you 50 coins and an HV lootbag? Where previous quests in the same chain give you a magnitude more coin? I don't see players doing the quest at that point caring about this. The mentioned quests are:
[Formaldehyde-AAAAAAAAAAAAAAAAAAADWQ==.json](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/pull/21034/files#diff-bceca5a30850ce7317076175d1cfc39f3ada8bfd33ece5f6f95cb4d3cec1c128)
[FormaldehydeCata-AAAAAAAAAAAAAAAAAAAJWw==.json](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/pull/21034/files#diff-4d6c94e3a82cd22416cfbc94eeaff27e67d3c2d6eaa69a069cfae582d0011a29)
[NitrogenTetroxid-AAAAAAAAAAAAAAAAAAAI9Q==.json](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/pull/21034/files#diff-61f23c4be26d08ca2b517cf53e1dc3a5bb37b69e2d7a50cd24f807e82d325511)
[11Dimethylhydraz-AAAAAAAAAAAAAAAAAAADWA==.json](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/pull/21034/files#diff-93dd51a2d62875501994442c413b1e98da7e3d739c5f9a6e775c729c23895755)

HFPTetraUro commented:
I have updated the layout to be more symmetrical/straight/consistently aligned, and added coin rewards to all steps.

Here are all the quests with their rewards. And just for the record, all of the early quests which give lootbags didn't give any before (Again, I stopped adding lootbags at the quest which actually requires you to be in IV, since HV lootbag doesn't mean anything there). You get more of those lootbags overall than before and they are more appropriately tiered to the beginning of the quest chain.


<img width="315" height="720" alt="image" src="https://github.com/user-attachments/assets/c24732d0-d88f-4062-b02c-3b46492bd774" />


<img width="1782" height="659" alt="image" src="https://github.com/user-attachments/assets/97dec572-9799-4e7f-ad29-7a740713c748" />
<img width="1788" height="564" alt="image" src="https://github.com/user-attachments/assets/ffed0170-e868-422a-82dc-b73acba66e53" />
<img width="1782" height="254" alt="image" src="https://github.com/user-attachments/assets/995d76ec-4717-4a37-8781-6eb2d692bdbc" />
<img width="1780" height="593" alt="image" src="https://github.com/user-attachments/assets/b9b98b8f-e5fb-49ad-b33b-cf338c0b8393" />
<img width="1774" height="654" alt="image" src="https://github.com/user-attachments/assets/88f747b3-9035-4d9a-b71c-746a82dfbb96" />
<img width="1766" height="567" alt="image" src="https://github.com/user-attachments/assets/336449be-dc3d-461c-85de-8e99a0d22d2e" />
<img width="1791" height="587" alt="image" src="https://github.com/user-attachments/assets/ca015bfd-b840-4391-8a45-c4be102c24ce" />
<img width="1780" height="564" alt="image" src="https://github.com/user-attachments/assets/87607c44-d71c-4d83-9637-8082ec634d99" />
<img width="1798" height="616" alt="image" src="https://github.com/user-attachments/assets/3a4e7329-5320-434f-b150-2caaf7cb05bd" />
<img width="1770" height="573" alt="image" src="https://github.com/user-attachments/assets/9c0c4fab-f335-4089-b66f-ffb4d74142b0" />
<img width="1780" height="556" alt="image" src="https://github.com/user-attachments/assets/ec9cb8ce-c807-4a54-9330-a668bf16e7b1" />
<img width="1792" height="560" alt="image" src="https://github.com/user-attachments/assets/6570561c-2909-4cb3-8a11-f45fc03f32fb" />
<img width="1789" height="577" alt="image" src="https://github.com/user-attachments/assets/ddfb25f9-79f1-4d19-bf04-9bf93ba180c8" />
<img width="1768" height="548" alt="image" src="https://github.com/user-attachments/assets/3df429d3-532f-4405-a835-dfe7961965fa" />
<img width="1784" height="702" alt="image" src="https://github.com/user-attachments/assets/780745f0-cab9-4451-a324-3516641aaf22" />
<img width="1803" height="585" alt="image" src="https://github.com/user-attachments/assets/19156c0d-38c2-4432-b115-b9efb2996098" />
<img width="1767" height="594" alt="image" src="https://github.com/user-attachments/assets/8552498f-ed88-4a11-9a26-e9f95b7836e5" />
<img width="1777" height="574" alt="image" src="https://github.com/user-attachments/assets/8ef8a0fc-22e9-4318-af86-e5a4fd51eb24" />
<img width="1799" height="549" alt="image" src="https://github.com/user-attachments/assets/707dbb57-b479-41b8-90e1-0572f39b8d8b" />
<img width="1783" height="592" alt="image" src="https://github.com/user-attachments/assets/5574df7f-9a4a-4253-8052-cf31e13b7db5" />

UltraProdigy reviewed (approved):
Grammar problems are fixed and all loot has been restored to the quests. If there are any further minor issues I can easily correct them in future patches but this should merge.

Dream-Master commented on config/betterquesting/DefaultQuests/Quests/SpaceRace-AAAAAAAAAAAAAAAAAAAAIg==/11Dimethylhydraz-AAAAAAAAAAAAAAAAAAADWA==.json:
```diff
@@ -36,40 +36,21 @@
           "Count:3": 1,
-          "Damage:2": 6,
-          "OreDict:8": "",
-          "id:8": "enhancedlootbags:lootbag"
```
This lootbags is still missing

Dream-Master commented on config/betterquesting/DefaultQuests/Quests/SpaceRace-AAAAAAAAAAAAAAAAAAAAIg==/Formaldehyde-AAAAAAAAAAAAAAAAAAADWQ==.json:
```diff
@@ -36,40 +36,21 @@
           "Count:3": 1,
-          "Damage:2": 6,
-          "OreDict:8": "",
-          "id:8": "enhancedlootbags:lootbag"
```
Same here

Dream-Master commented on config/betterquesting/DefaultQuests/Quests/SpaceRace-AAAAAAAAAAAAAAAAAAAAIg==/H8N4C2O4GreenRoc-AAAAAAAAAAAAAAAAAAAAkw==.json:
```diff
@@ -44,40 +44,29 @@
           "Count:3": 1,
-          "Damage:2": 42,
-          "OreDict:8": "",
-          "id:8": "enhancedlootbags:lootbag"
```
Or here

Dream-Master commented on config/betterquesting/DefaultQuests/Quests/SpaceRace-AAAAAAAAAAAAAAAAAAAAIg==/NitrogenTetroxid-AAAAAAAAAAAAAAAAAAAI9Q==.json:
```diff
@@ -44,40 +44,21 @@
           "Count:3": 1,
-          "Damage:2": 6,
-          "OreDict:8": "",
-          "id:8": "enhancedlootbags:lootbag"
```
Or here

Nikolay-Sitnikov commented:
In terms of continuing to improve the quest text:
- Instead of calling everything "ingredient", we could call the ingredients "fuel" & "oxidizer". (Nitric Acid & Nitrogen Tetroxide are the oxidizers, all the hydrazine derivatives are the fuels.) (Although the completed mixture is also called "fuel", so that might now work.)
- There's a sentence which starts off with an "And either Rare Earth (II) ..." in the Laurenium Casing quest. It should be changed to "You'll also need either Rare Earth (II) ..."
- The reason for the numbers in the names of the chemicals is that it tells you where the modifiers are placed on the base molecule. So "2-ethylanthraquinone" is a molecule of anthraquinone with one ethyl group attached to atom 2. Although I do feel like the original quest text feels friendlier, so maybe we don't change it?

UltraProdigy reviewed (approved):
All problems fixed and requested changes resolved.
